src/main/java/uk/gov/companieshouse/api/accounts/model/ixbrl/period/Period.java

```diff
@@ -13,35 +16,98 @@
     @JsonProperty("previous_period_end_on")
     private String previousPeriodEndsOn;
 
+    @JsonProperty("current_period_start_on_formatted")
+    private String currentPeriodStartOnFormatted;
+    @JsonProperty("current_period_end_on_formatted")
+    private String currentPeriodEndsOnFormatted;
+
+    @JsonProperty("current_period_bs_date")
+    private String currentPeriodBSDate;
+    @JsonProperty("previous_period_bs_date")
+    private String previousPeriodBSDate;
+
+    private AccountsDates accountsDates;
+
+    public Period(AccountsDates accountsDates) {
+
+        this.accountsDates = accountsDates;
+
+    }
+
+    public String getCurrentPeriodBSDate() {
+        return currentPeriodBSDate;
+    }
+
+    public void setCurrentPeriodBSDate(String currentPeriodStartOn, String currentPeriodEndsOn, boolean isSameYear) {
+        currentPeriodBSDate = accountsDates.generateBalanceSheetHeading(currentPeriodStartOn, currentPeriodEndsOn,
+                isSameYear(currentPeriodStartOn, currentPeriodEndsOn));
+    }
+
+    public String getPreviousPeriodBSDate() {
+        return previousPeriodBSDate;
+    }
+
+    public void setPreviousPeriodBSDate(String previousPeriodStartOn, String previousPeriodEndsOn, boolean isSameYear) {
+        previousPeriodBSDate = accountsDates.generateBalanceSheetHeading(previousPeriodStartOn, previousPeriodEndsOn,
+                isSameYear(currentPeriodStartOn, currentPeriodEndsOn));
```

Comment:
Just double checking... isSameYear(currentPeriodStartOn, currentPeriodEndsOn) is correct in setPreviousPeriodBSDate?